oneapp-server/dao/ApplicationDao.js

```diff
@@ -0,0 +1,69 @@
+const { SQLDataSource } = require('datasource-sql');
+const oracledb = require('oracledb');
+const { toUpper } = require('lodash');
+const { OneAppError, OneAppAuthenticationError } = require('../utils/OneAppError');
+const logger = require('../logger.config');
+
+class ApplicationDao extends SQLDataSource {
+  async sendApplication(APPLICATION_ID, DISCLAIMER_UNDERSTOOD) {
+    try {
+      const con = await this.knex.client.pool.acquire().promise;
+      const response = await this.knex.client.transaction(async (tx) => {
+        const bindVars = {
+          app_id: { dir: oracledb.BIND_IN, val: APPLICATION_ID },
+          disclaimer: { dir: oracledb.BIND_IN, val: DISCLAIMER_UNDERSTOOD },
+          info: { dir: oracledb.BIND_OUT, type: oracledb.DB_TYPE_CURSOR },
+        };
+        return tx.raw('begin OA_PKG_APP.SP_SEND_APPLICATION(:app_id, :disclaimer, :info); end;', bindVars);
+      }, { connection: con });
+
+      try {
+        const result = await response[0].getRow();
+        return result;
+      } finally {
+        response[0].close();
+      }
+    } catch (err) {
+      logger.error(err);
+      throw new OneAppError('Failed to send application', 't247');
+    }
+  }
+
+  async fetchConfirmationDetails(APPLICATION_NUMBER, USER_ID) {
+    try {
+      const con = await this.knex.client.pool.acquire().promise;
+      const response = await this.knex.client.transaction(async (tx) => {
+        // Don't allow access to applications the user does not own
+        const userApplication = await tx
+          .select('APPLICATION_NUMBER')
+          .from('OA_APP_APPLICATION_HEADER')
+          .whereRaw('regexp_like(USER_ID, ?, \'i\') and APPLICATION_NUMBER = ?', [`^${toUpper(USER_ID)}$`, APPLICATION_NUMBER]) // Check against case-insensitive USER_ID
+          .first();
+        if (userApplication == null) {
+          throw new OneAppAuthenticationError('You do not have access to this data.');
+        }
+
+        const bindVars = {
+          app_id: { dir: oracledb.BIND_IN, val: APPLICATION_NUMBER },
+          confirmation: { dir: oracledb.BIND_OUT, type: oracledb.DB_TYPE_CURSOR },
+        };
+        return tx.raw('begin OA_PKG_APP.GET_CONFIRMATION_DETAILS(:app_id, :confirmation); end;', bindVars);
```

Comment:
Any reason you're using this stored procedure instead of [OA_PKG_GEN's](https://github.com/newjersey/dhs-oneapp-legacy/blob/080a57380e9dedd834d2b6de141d7a031825159e/OneAppStoredProceduresFunctions/OA_PKG_GEN.sql#L452)? I can't find this call in OA_PKG_APP, and the legacy code seems to call OA_PKG_GEN?

Reply:
Ooo, good call. I had copied / pasted from send application and had the wrong package prefix. I also hadn't tested this stored procedure since the copy/paste.

I locally disabled the user protection (since I can't test `sendApplication` yet) and verified this stored procedure is now working and calling the correct package. I also made a few type changes on the output to make for a cleaner API.